whois/src/main.rs

```diff
@@ -90,16 +120,25 @@ impl Telemetry {
 	}
 }
 
-fn save_node_id(node: &AddedNode, network_id: String, holder: &mut Option<FeedNodeId>) {
-	let node_network_id = node.details.network_id.clone();
-	if node_network_id == Some(network_id) {
-		*holder = Some(node.node_id);
+fn save_node_id(node: &AddedNode, authority_key: AccountId32, holder: &mut Option<FeedNodeId>) {
```

Comment:
Cannot we just return an `Option<FeedNodeId>`, as the current approach reminds me the plain C one.